pandas/core/base.py

```diff
@@ -87,6 +88,14 @@ def __sizeof__(self):
         # object's 'sizeof'
         return super().__sizeof__()
 
+    def _ensure_type(self: T, obj) -> T:
+        """Ensure that an object has same type as self.
+
+        Used by type checkers.
+        """
+        assert isinstance(obj, type(self)), type(obj)
```

Comment:
This line breaks any code that creates a subclass of the `pandas.core.frame.DataFrame`. `issubclass` should be used instead.

Reply:
Yes, see also https://github.com/pandas-dev/pandas/issues/31925 (maybe repeat your comment there). We should fix this for 1.0.2